src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -515,6 +517,50 @@ impl AstDisplay for AlterObjectRenameStatement {
 }
 impl_display!(AlterObjectRenameStatement);
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum AlterIndexOptionsList {
+    Set(Vec<SqlOption>),
+    Reset(Vec<Ident>),
+}
+
+/// `ALTER INDEX ... {RESET, SET}`
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AlterIndexOptionsStatement {
+    pub index_name: ObjectName,
+    pub if_exists: bool,
+    pub options: AlterIndexOptionsList,
+}
+
+impl AstDisplay for AlterIndexOptionsStatement {
+    fn fmt(&self, f: &mut AstFormatter) {
+        f.write_str("ALTER INDEX ");
+        f.write_node(&self.index_name);
+        f.write_str(" ");
+        if self.if_exists {
+            f.write_str("IF EXISTS ");
+        }
+
+        match &self.options {
+            AlterIndexOptionsList::Set(options) => {
+                if !options.is_empty() {
+                    f.write_str("SET (");
+                    f.write_node(&display::comma_separated(&options));
+                    f.write_str(")");
+                }
```

Comment:
Conditioning this on `options.is_empty` seems a bit weird—there's syntactically required to be at least one, right? Maybe an assert instead? As written you could end up with `ALTER INDEX foo` which doesn't seem right.

Reply:
removed the conditional. i think beyond that the assert isn't really required